synapse/logging/_structured.py

```diff
@@ -89,14 +88,7 @@ def __call__(self, event: dict) -> None:
         context = current_context()
 
         # Copy the context information to the log event.
-        if context is not None:
-            context.copy_to_twisted_log_entry(event)
-        else:
-            # If there's no logging context, not even the root one, we might be
-            # starting up or it might be from non-Synapse code. Log it as if it
-            # came from the root logger.
-            event["request"] = None
-            event["scope"] = None
+        context.copy_to_twisted_log_entry(event)
```

Comment:
`current_context` no longer returns `None`, but `_SentinelContext` (#7120), which implements `copy_to_twisted_log_entry` with the removed logic.